gui/src/ScenarioViewer.tsx

```diff
@@ -0,0 +1,539 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import Grid from "@mui/material/Grid";
+import IconButton from "@mui/material/IconButton";
+import Dialog from "@mui/material/Dialog";
+import TextField from "@mui/material/TextField";
+import Typography from "@mui/material/Typography";
+import {
+    FlagOutlined,
+    Close,
+    DeleteOutline,
+    Add,
+    Send,
+    CheckCircle,
+    Cancel,
+    ArrowForwardIosSharp,
+} from "@mui/icons-material";
+
+import {
+    Accordion,
+    AccordionDetails,
+    AccordionSummary,
+    AccordionSummaryProps,
+    Autocomplete,
+    Chip,
+    Divider,
+    styled,
+} from "@mui/material";
+import { FlagSx, Property, ScenarioFull } from "./utils";
+
+interface ScenarioViewerProps {
+    id?: string;
+    expandable?: boolean;
+    expanded?: boolean;
+    label?: string;
+    buttonToSubmitScenario?: boolean;
+    buttonToDeleteScenario?: boolean;
+    primary?: boolean;
+    configIdNotEditable?: boolean;
+    frequencyNotEditable?: boolean;
+    cycleNotEditable?: boolean;
+    updateVarName?: string;
+    tags?: Array<string>;
+    scenario?: ScenarioFull;
+    onSubmit: string;
+    error?: string;
+}
+
+interface PipelinesRowProps {
+    action: string;
+    number: number;
+    value: string;
+}
+
+const MainBoxSx = {
+    overflowY: "auto",
+};
+
+const FieldNoMaxWidth = {
+    maxWidth: "none",
+};
+
+const AccordionSummarySx = { fontSize: "0.9rem" };
+const IconButtonSx = { p: 0 };
+const ChipSx = { ml: 1 };
+const IconPaddingSx = { padding: 0 };
+const DeleteIconSx = { height: 50, width: 50, p: 0 };
+
+const tagsAutocompleteSx = {
+    "& .MuiOutlinedInput-root": {
+        padding: "3px 15px 3px 3px !important",
+    },
+    maxWidth: "none",
+};
+
+const PipelineRow = ({ action, number, value }: PipelinesRowProps) => {
+    const [pipeline, setPipeline] = useState<string>(value);
+    const [focus, setFocus] = useState(false);
+
+    const onFieldBlur = useCallback((e: any) => {
+        setPipeline(e.target.value);
+        setFocus(false);
+    }, []);
+
+    const index = number + 1;
+    return (
+        <Grid item xs={12} container justifyContent="space-between">
+            <Grid item container xs={10}>
+                {action === "EDIT" && (
+                    <TextField
+                        data-name={`pipe${index}`}
+                        label={"Pipeline " + index}
+                        variant="outlined"
+                        name={`pipeline${index}`}
+                        value={pipeline}
+                        onBlur={onFieldBlur}
+                        sx={FieldNoMaxWidth}
+                        onFocus={(e) => setFocus(true)}
+                        fullWidth
+                        InputProps={{
+                            endAdornment: focus && (
```

Comment:
will they disappear when you click on them ?
I thought it was another field getting the focus that removed it from the others ?

Reply:
Could you put more explanation? I am not sure if I understood this question. :) @FredLL-Avaiga